src/opencensus.erl

```diff
@@ -91,15 +139,40 @@ finish_span(Span=#span{start_time=StartTime}) ->
 
 %%--------------------------------------------------------------------
 %% @doc
-%% Starts a new span as a child of a existing span, using the parents
-%% Trace ID and setting the childs parent to the parents Span ID
+%% Return the current trace context for a span, to be used for
+%% propagation across process boundries.
 %% @end
 %%--------------------------------------------------------------------
--spec child_span(unicode:chardata(), maybe(span())) -> maybe(span()).
-child_span(_Name, undefined) ->
+context(undefined) ->
     undefined;
-child_span(Name, #span{trace_id = TraceId, span_id = ParentId}) ->
-    start_span(Name, TraceId, ParentId).
+context(#span{trace_id=TraceId,
+              span_id=SpanId}) ->
+    #trace_context{trace_id=TraceId,
+                   span_id=SpanId}.
+
+%%--------------------------------------------------------------------
+%% @doc
+%% Put an attribute (a key/value pair) in the attribute map of a span.
+%% If the attribute already exists it is overwritten with the new value.
+%% @end
+%%--------------------------------------------------------------------
+-spec put_attribute(unicode:chardata(), attribute_value(), maybe(span())) -> maybe(span()).
+put_attribute(_Key, _Value, undefined) ->
+    undefined;
+put_attribute(Key, Value, Span=#span{attributes=Attributes}) ->
```

Comment:
same as the ocp version, would it make sense to guard against bad attributes here since we can?